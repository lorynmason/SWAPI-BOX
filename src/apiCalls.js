export const fetchScroll = async () => {
 const url = 'https://swapi.co/api/films';
 const response = await fetch(url);
 if(response.ok) {
  const filmsData = await response.json();
  return filmsData
  } else {
    throw new Error('Yoda- ERROR has detected, API returned okay NOT')
  }
}

export const fetchCharacters = async () => {
  const url = 'https://swapi.co/api/people'
  const response = await fetch(url);
  if(response.ok) {
  const characterData = await response.json()
  return characterData
  } else {
    throw new Error('Chewbacca- AGERUYEHSFG: translation, Error, API returned not okay')
  }
}

export const fetchNestedInfo = (characterData) => {
  const sadPromises = characterData.results.map( async (character) => {
      const response = await fetch(character.homeworld)
      const homeworlds = await response.json()
      const response2 =  await fetch(character.species)
      const species = await response2.json()
      return {homeworld: homeworlds.name,
              population: homeworlds.population,
              species: species.name,
              name: character.name}
  } )
  return Promise.all(sadPromises)
}

export const fetchVehicles = async () => {
  const url = 'https://swapi.co/api/vehicles'
  const response = await fetch(url);
  const vehicleData = await response.json()
  return vehicleData
}







