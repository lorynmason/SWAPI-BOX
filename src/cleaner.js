export const cleanFilmsData = (data) => {
  const films = data.results.map((film) => {
    return {
      title: film.title,
      text: film.opening_crawl,
      date: film.release_date
    }
  })
  return films
}

export const cleanCharacterData = (data) => {
  return data.map((character) => {
    return {
      id: character.name,
      category: 'characters',
      info: {
        Name: character.name,
        Homeworld: `Homeworld:  ${character.homeworld}`,
        Population: `Population:  ${character.population}`,
        Species: `Species:  ${character.species}`
      }
    }
  })
}

export const cleanVehiclesData = (data) => {
  return data.results.map((vehicle) => {
    return {
      id: vehicle.name,
      category: 'vehicles',
      info: {
        Name: vehicle.name,
        Model: `Model:  ${vehicle.model}`,
        Class: `Class:  ${vehicle.vehicle_class}`,
        Passengers: `Passengers:  ${vehicle.passengers}`
      }
    }
  })
}

export const cleanPlanetData = (data) => {
  return data.map((planet) => {
    if (planet.residents.length <= 0) {
      planet.residents = 'No known occupants'
    }
    return {
      id: planet.name,
      category: 'planets',
      info: {
        Name: planet.name,
        Terrain: `Terrain:  ${planet.terrain}`,
        Population: `Population:  ${planet.population}`,
        Climate: `Climate:  ${planet.climate}`,
        Residents: `Residents:  ${planet.residents}`
      }
    }
  })
}
