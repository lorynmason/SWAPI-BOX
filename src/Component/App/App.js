import React, { Component } from 'react';
import '../styles/main.scss';
import * as API from '../../apiCalls.js';
import * as Cleaner from '../../cleaner.js';
import Nav from '../Nav/Nav.js';
import Splash from '../Splash/Splash.js';
import CardContainer from '../CardContainer/CardContainer.js';
import Header from '../Header/Header.js';

class App extends Component {
  constructor() {
    super()
    this.state = {
      film: {},
      favorites: [],
      activePage: '',
      splash: true,
      characters: [],
      vehicles: []
    }
  }
  async componentDidMount() {
    const filmsData = await API.fetchScroll()
    const film = Cleaner.cleanFilmsData(filmsData)
    this.setState({
      film
    })
  }

  async componentDidUpdate() {
    const { characters, activePage, vehicles } = this.state;
    if(activePage === 'characters' && characters.length === 0) {
      const  characterData= await API.fetchCharacters()
      const characters = await API.fetchNestedInfo(characterData)
      this.setState({characters})
    }
    if(activePage === 'vehicles' && vehicles.length === 0) {
      const vehicleData = await API.fetchVehicles()
      const vehicles = Cleaner.cleanVehiclesData(vehicleData)
      this.setState({vehicles})
    }
    
  }

  exitSplash = () => {
    this.setState({
      splash: false,
      activePage: 'home'
    })
  }

  changePage=(str)=> {
    this.setState({
      activePage: str
    })
  }
  
  render() {
    if(this.state.splash) {
      return (
        <div className="splash">
        <Splash exitSplash={this.exitSplash} film={this.state.film}/>
        </div>
      );
    } else {
      return (
        <div className="App">
          <Header/>
          <Nav favorites={this.state.favorites} 
               changePage={this.changePage}/>
          <CardContainer  activePage={this.state.activePage}              characters={this.state.characters}
                          vehicles={this.state.vehicles}/>
        </div>
      )
    }
  }
}

export default App;
