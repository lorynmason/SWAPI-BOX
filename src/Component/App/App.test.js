import React from 'react'
import { shallow } from 'enzyme'
import App from './App'
import * as API from '../../apiCalls'
import * as Cleaner from '../../cleaner'

describe('App', () => {
  let wrapper
  const mockFunc = jest.fn()
  API.fetchData = mockFunc

  beforeEach(() => {
    wrapper = shallow(<App />)
  })

  describe('componentDidMount', () => {
    it('should call getLocalStorage, API.fetchData, Cleaner.cleanFilmsData, addLocalStorage', async() => {
      wrapper.getLocalStorage = mockFunc
      Cleaner.cleanFilmsData = mockFunc
      wrapper.addLocalStorage = mockFunc
      expect(wrapper.getLocalStorage).toHaveBeenCalled()
      expect(API.fetchData).toHaveBeenCalled()
      expect(Cleaner.cleanFilmsData).toHaveBeenCalled()
      expect(wrapper.addLocalStorage).toHaveBeenCalled()

    })
  })

  describe('setCharacterData', () => {
    it('Should call API function', async () => {
      const obj = {results: {}}
      API.fetchData = jest.fn(() => obj)
      API.getMoreCharacterData = jest.fn(() => 'Luke Skywalker')

      await wrapper.instance().setCharacterData()
      expect(API.fetchData).toHaveBeenCalled()
      expect(API.getMoreCharacterData).toHaveBeenCalled()
      const result = wrapper.state('characters')
      expect(result).toEqual('Luke Skywalker')
    })
  })

  describe('setVehicleData', () => {
    it('Should call API function', async () => {
      Cleaner.cleanVehiclesData = jest.fn(() => 'Sand Crawler')

      await wrapper.instance().setVehicleData()
      expect(API.fetchData).toHaveBeenCalled()
      expect(Cleaner.cleanVehiclesData).toHaveBeenCalled()
      const result = wrapper.state('vehicles')
      expect(result).toEqual('Sand Crawler')
    })
  })

  describe('setPlanetData', () => {
    it('should setState of planets to planets', async () => {
      API.fetchData = mockFunc
      API.fetchNestedInfoPlanets = mockFunc
      Cleaner.cleanPlanetData = jest.fn(() => 'Hoth')
      await wrapper.instance().setPlanetData()
      expect(API.fetchData).toHaveBeenCalled()
      expect(API.fetchNestedInfoPlanets).toHaveBeenCalled()
      const result = wrapper.state('planets')
      expect(result).toEqual('Hoth')
    })
  })

  describe('changePage', () => {
    it('should setState of activePage to whatever is passed in', () => {
      const str = 'characters'
      wrapper.instance().changePage(str)
      expect(wrapper.state('activePage')).toEqual('characters')
      wrapper.setState({activePage: 'splash'})
    })
  })

  describe('toggleFavorites', () => {
    it('should find card by the id', async () => {
      const expected = ['Luke Skywalker']
      expect(wrapper.state('favorites')).toEqual([])
      await wrapper.setState({ activePage: 'characters' })
      await wrapper.instance().toggleFavorites('Luke Skywalker')
      expect(wrapper.state('favorites')).toEqual(expected)
    })
  })

  describe('render', () => {
    it('should match the snapshot if activePage = splash', () => {
      wrapper.setState({ activePage: 'splash', splash: [] })
      expect(wrapper).toMatchSnapshot()
    })

    it('should match the snapshot if activePage !== splash', () => {
      wrapper.setState({ activePage: 'characters' })
      expect(wrapper).toMatchSnapshot()
    })
  })
})
