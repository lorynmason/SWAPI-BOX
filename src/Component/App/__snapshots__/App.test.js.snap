// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App render should match the snapshot if activePage !== splash 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Header />,
        <Nav
          changePage={[Function]}
          favorites={
            Array [
              "Luke Skywalker",
            ]
          }
        />,
        <Switch>
          <Route
            component={[Function]}
            exact={true}
            path="/"
          />
          <Route
            path="/characters"
            render={[Function]}
          />
          <Route
            path="/planets"
            render={[Function]}
          />
          <Route
            path="/vehicles"
            render={[Function]}
          />
          <Route
            path="/favorites"
            render={[Function]}
          />
        </Switch>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "changePage": [Function],
          "favorites": Array [
            "Luke Skywalker",
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />,
            <Route
              path="/characters"
              render={[Function]}
            />,
            <Route
              path="/planets"
              render={[Function]}
            />,
            <Route
              path="/vehicles"
              render={[Function]}
            />,
            <Route
              path="/favorites"
              render={[Function]}
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "path": "/characters",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "path": "/planets",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "path": "/vehicles",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "path": "/favorites",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header />,
          <Nav
            changePage={[Function]}
            favorites={
              Array [
                "Luke Skywalker",
              ]
            }
          />,
          <Switch>
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />
            <Route
              path="/characters"
              render={[Function]}
            />
            <Route
              path="/planets"
              render={[Function]}
            />
            <Route
              path="/vehicles"
              render={[Function]}
            />
            <Route
              path="/favorites"
              render={[Function]}
            />
          </Switch>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "changePage": [Function],
            "favorites": Array [
              "Luke Skywalker",
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                component={[Function]}
                exact={true}
                path="/"
              />,
              <Route
                path="/characters"
                render={[Function]}
              />,
              <Route
                path="/planets"
                render={[Function]}
              />,
              <Route
                path="/vehicles"
                render={[Function]}
              />,
              <Route
                path="/favorites"
                render={[Function]}
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "path": "/characters",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "path": "/planets",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "path": "/vehicles",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "path": "/favorites",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`App render should match the snapshot if activePage = splash 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "changePage": [Function],
      "splash": Array [],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "changePage": [Function],
        "splash": Array [],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
